### PR TITLE
Allow a POS seat to overwrite its IP country on POS Session with a ma…

### PR DESCRIPTION
**Pull request by ithiame**
…nually selected country #1186

### CODE REVIEW

File: src/lib/types/Session.ts [hunk 1]
```diff
@@ -16,4 +16,5 @@ export interface Session extends Timestamps {
 		name: string;
 		id: string;
 	}>;
+	clientIp?: string;
```
Comment (coyotte508) [inline src/lib/types/Session.ts:19]: Let's store `pos?: { countryCodeOverwrite?: ...}` instead

Let's not store a client ip in the session or mess with ips.

The `ipFromCountry` is a nice bit of code, and was fun to write probably, but boring code/simplest solution is better for long term :pray:

Comment (ithiame) [reply src/lib/types/Session.ts:19]: So what do you mean ? There's no need to store the IP ?
If so how are we gonna the clientIP ?


(I took Inspiration on `countryFromIP`)

File: src/hooks.server.ts [hunk 2]
```diff
@@ -216,6 +216,10 @@ const handleGlobal: Handle = async ({ event, resolve }) => {
 		event.locals.email = session.email;
 		event.locals.npub = session.npub;
 		event.locals.sso = session.sso;
+		event.locals.clientIp = session.clientIp;
+		event.locals.countryCode = event.locals.clientIp
+			? countryFromIp(event.locals.clientIp)
+			: undefined;
```
Comment (coyotte508) [inline src/hooks.server.ts:222]:
```suggestion
		if (session.pos?.countryCodeOverride) {
			event.locals.countryCode = session.pos.countryCodeOverride;
		};
```

I think that's what we want @ithiame . No need to store a fake IP, only the fake countrycode

File: src/routes/(app)/pos/+page.svelte [hunk 3]
```diff
@@ -16,6 +26,23 @@
...
 		>{t('pos.adminInterface')}</a
 	>
+	<form action="?/overwrite" method="POST" on:submit={handleSubmit}>
+		<label class="checkbox-label">
+			<input type="checkbox" class="form-checkbox" name="overwriteIP" bind:checked={overwriteIP} />
+			Overwrite IP country for this POS session with selected country
+		</label>
+		<label class="form-label col-span-3">
+			Overwrite session IP country (current value {data.countryCode})
+			<select name="countryCode" class="form-input" required bind:value={data.countryCode}>
```
Comment (coyotte508) [inline src/routes/(app)/pos/+page.svelte:36]:
```suggestion
			<select name="countryCode" class="form-input" required value={data.countryCode}>
```

no?

**Review by coyotte508 — APPROVED** [review]

**Comment by Tirodem** [comment]
Nice, it works.
But, and that might be because of me : how do we remove the overwrite ? It is possible or shall we overwrite with a new country ?
I suggest to add an option "(Remove ipcountry overwrite)" at the beginning of the country list, or add a button "Remove overwrite".
@coyotte508 , WDYT ?

**Comment by coyotte508** [comment]
Yes agree. we could show the "remove overwrite" button when it's already overwritten

File: src/routes/(app)/pos/+page.server.ts [hunk 4]
```diff
@@ -32,7 +36,8 @@ export const load = async (event) => {
 				})) || [],
 			status: order.status
 		})),
-		countryCode: event.locals.countryCode
+		countryCode: event.locals.countryCode,
+		session: pojo(session)
```
Comment (coyotte508) [inline src/routes/(app)/pos/+page.server.ts:40]: Please send only the needed data, for security reasons. Not the whole session

**Review by coyotte508 — APPROVED** [review]

**Pull request merged**